src/modals/metadataData/EditMetadataDataModal.vue

```diff
@@ -0,0 +1,235 @@
+
+			<div v-if="success > -1">
+				<NcNoteCard v-if="success" type="success" heading="Success!">
+					<p>Successfully updated metadata property {{ store.metadataDataKey }}</p>
+				</NcNoteCard>
+				<NcNoteCard v-if="!success" type="error" heading="Error!">
+					<p>Something went wrong</p>
+				</NcNoteCard>
+
+				<NcButton
+					type="primary"
+					@click="closeModal">
+					Close
+				</NcButton>
+			</div>
+		</div>
+	</NcModal>
+</template>
+
+<script>
+import {
+	NcButton,
+	NcModal,
+	NcTextField,
+	NcSelect,
+	NcCheckboxRadioSwitch,
+	NcLoadingIcon,
+	NcNoteCard,
+} from '@nextcloud/vue'
+
+export default {
+	name: 'EditMetadataDataModal',
+	components: {
+		NcModal,
+		NcTextField,
+		NcSelect,
+		NcCheckboxRadioSwitch,
+		NcButton,
+		NcLoadingIcon,
+		NcNoteCard,
+	},
+	data() {
+		return {
+			metadata: {
+				title: '',
+				description: '',
+				version: '',
+				properties: {},
+				_schema: '',
+				_id: '',
+				id: '',
+			},
+			dataKey: '',
+			typeOptions: {
+				inputLabel: 'Type',
+				multiple: false,
+				options: [
+					'string',
+					'boolean',
+					'object',
+					'array',
+				],
+			},
+			loading: false,
+			success: -1,
+			hasUpdated: false,
+			metadataLoading: false,
+		}
+	},
+	updated() {
+		if (store.modal === 'editMetadataDataModal' && this.hasUpdated) {
+			if (this.dataKey !== store.metadataDataKey) this.hasUpdated = false
+		}
+		if (store.modal === 'editMetadataDataModal' && !this.hasUpdated) {
+			this.metadata = {
+				...store.metaDataItem,
+				properties: JSON.parse(store.metaDataItem.properties),
+			}
+			this.metadata.properties[store.metadataDataKey] = store.getMetadataPropertyKeys(store.metadataDataKey)
+			this.fetchData(store.metaDataItem.id)
+
+			this.dataKey = store.metadataDataKey
+			this.hasUpdated = true
+		}
+	},
+	methods: {
+		fetchData(id) {
+			this.metadataLoading = true
+			fetch(
+				`/index.php/apps/opencatalogi/api/metadata/${id}`,
+				{
+					method: 'GET',
+				},
+			)
+				.then((response) => {
+					response.json().then((data) => {
+						store.metaDataItem = data
+
+						this.metadata = {
+							...store.metaDataItem,
+							properties: JSON.parse(store.metaDataItem.properties),
+						}
+						this.metadata.properties[store.metadataDataKey] = store.getMetadataPropertyKeys(store.metadataDataKey)
+					})
+					this.metadataLoading = false
+				})
+				.catch((err) => {
+					console.error(err)
+					this.metadataLoading = false
+				})
+		},
+		closeModal() {
+			store.modal = false
+		},
+		updateMetadata(id) {
+			this.loading = true
+			fetch(
+				`/index.php/apps/opencatalogi/api/metadata/${id}`,
+				{
+					method: 'PUT',
+					headers: {
+						'Content-Type': 'application/json',
+					},
+					body: JSON.stringify({
+						...this.metadata,
+						properties: JSON.stringify(this.metadata.properties),
+					}),
+				},
+			)
+				.then((response) => {
+					this.loading = false
+					this.success = 1
+					setTimeout(() => this.closeModal(), 3000)

```

Comment:
Je moet in de time out ook wel de succes status terug zetten anders werkt het formulier in de volgende iteratie niet. Tevens mis ik het verversen van de metadatalist dus ik gok als ik de code lees dat de gebruiker niet ziet dat de aanpassing is verwerkt.